Userland/Libraries/LibGUI/SettingsWindow.cpp

```diff
@@ -57,13 +57,19 @@ ErrorOr<NonnullRefPtr<SettingsWindow>> SettingsWindow::create(DeprecatedString t
     window->m_ok_button = TRY(button_container->try_add<GUI::DialogButton>("OK"));
     window->m_ok_button->on_click = [window = window->make_weak_ptr<SettingsWindow>()](auto) {
         window->apply_settings();
-        GUI::Application::the()->quit();
+        if (window->m_close_program_after_completion == CloseProgramAfterCompletion::Yes)
+            GUI::Application::the()->quit();
+        else
+            window->close();
```

Comment:
I think you might be able to just change this to always `window->close()`, and it will quit the application automatically if it was the only window. Then you wouldn't need `CloseProgramAfterCompletion`.